app/menu.ts

```diff
@@ -80,11 +79,10 @@ export default class MenuBuilder {
           }
         }
       ]
-    };
-    const subMenuEdit: MenuItemConstructorOptions = {
+    } as MenuItemConstructorOptions;
```

Comment:
I don't really like this, this will treat every kind of menu definition as a proper one. E.g.
```
const faultSubMenuAbout = {
labelWithType: 'Electron';
suubMenu: [],
} as MenuItemConstructorOptions;
```
No ts warnings:(

How about declaring a darwin specific type?
```
interface DarwinMenuItemConstructorOptions extends MenuItemConstructorOptions{
  selector?: string;
  submenu?: (DarwinMenuItemConstructorOptions[]) | (Menu);
}

const faultySubMenuAbout: DarwinMenuItemConstructorOptions {
typeInLable: 'Electron';
}
```

Reply:
I like this solution a lot more!